### PR TITLE
Replace publishing to Bintray with Maven Central

### LINKED ISSUE #1227: Drop Bintray and JCenter
Reason: https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

There's a blocker (trove 4j only available thru JCenter):

> Execution failed for task ':sentry-android-core:extractDebugAnnotations'.
> > Could not resolve all files for configuration ':sentry-android-core:lintClassPath'.
>    > Could not find org.jetbrains.trove4j:trove4j:20160824.
>      Searched in the following locations:
>        - https://dl.google.com/dl/android/maven2/org/jetbrains/trove4j/trove4j/20160824/trove4j-20160824.pom
>        - https://repo.maven.apache.org/maven2/org/jetbrains/trove4j/trove4j/20160824/trove4j-20160824.pom
>      Required by:
>          project :sentry-android-core > com.android.tools.lint:lint-gradle:27.1.2 > com.android.tools:sdk-common:27.1.2
>          project :sentry-android-core > com.android.tools.lint:lint-gradle:27.1.2 > com.android.tools.external.com-intellij:intellij-core:27.1.2

Replace the Bintray plugin: https://dl.bintray.com/maranda/maven/ by https://docs.gradle.org/current/userguide/publishing_maven.html

requires `POM` rewriting, https://central.sonatype.org/pages/requirements.html#sufficient-metadata

requires `signing artifacts` https://docs.gradle.org/current/userguide/publishing_signing.html

This could be followed by https://github.com/getsentry/sentry-java/issues/1063

Fix https://github.com/getsentry/sentry-java/tree/main/scripts as bintray won't be used anymore

https://chris.banes.dev/publishing-to-maven-central/

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Replace publishing to Bintray with Maven Central.
- dropped dependency to bintray-release Gradle plugin.
- added packages signing
- adapted release scripts to publish to OSSRH repository instead of bintray (not tested)

Ps: this allows us publishing to MC via Gradle plugin, not via CLI, yet

_#skip-changelog_

## :bulb: Motivation and Context
Fixes #1227

## :pencil: Checklist

- [X] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [X] No breaking changes

### CODE REVIEW

**Review by maciejwalkowiak — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -11,18 +11,19 @@ compile:
 	./gradlew build
 
 # do a dry release (like a local deploy)
+# remember to add the -SNAPSHOT suffix to the version
```
Comment (bruno-garcia) [inline Makefile:14]: remembering is hard, can we check for that before running this task? Or not really a big deal if we add a non SNAPSHOT version to local maven?

File: sentry-spring/src/main/java/io/sentry/spring/tracing/SentrySpanAdvice.java [hunk 2]
```diff
@@ -26,6 +26,7 @@ public SentrySpanAdvice(final @NotNull IHub hub) {
     this.hub = Objects.requireNonNull(hub, "hub is required");
   }
 
+  @SuppressWarnings("deprecation")
```
Comment (bruno-garcia) [inline sentry-spring/src/main/java/io/sentry/spring/tracing/SentrySpanAdvice.java:29]: Accidentally added?

**Review by bruno-garcia — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (marandaneto) [reply Makefile:14]: this is already part of our release manual and ideally should be automatized by Craft which is not the focus of this PR

File: sentry-spring/src/main/java/io/sentry/spring/tracing/SentrySpanAdvice.java (hunk 2 above)
Comment (marandaneto) [reply sentry-spring/src/main/java/io/sentry/spring/tracing/SentrySpanAdvice.java:29]: no, See https://github.com/getsentry/sentry-java/pull/1320/files#r597775167 (and other comments maybe too for context)

**Pull request merged**